### PR TITLE
Open link - Nguyen

### LINKED ISSUE #108: Open other apps
As a NetworkBook user, I wish to be able to open the relevant social media platform from the NetworkBook app so that I can quickly contact them on social media.

As a NetworkBook user, I wish to be able to open my email app from my NetworkBook app, so that I can quickly send an email when needed.

### PR DESCRIPTION
**Pull request by nknguyenhc**

### CODE REVIEW

**Comment by nknguyenhc** [comment]
For codecov report:
* Assert statements are falsely reported as partially tested. 
Test cases have been added to ensure that they throw assertion errors
when the condition is not met.
* Opening link is manually tested. 
Credits to @Singa-pirate for testing on Ubuntu and
@awhb for testing on Mac OS.

File: src/main/java/networkbook/logic/commands/OpenLinkCommand.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+     */
+    public OpenLinkCommand(Index personIndex, Index linkIndex) {
+        requireAllNonNull(personIndex, linkIndex);
+        this.personIndex = personIndex;
+        this.linkIndex = linkIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
```
Comment (Singa-pirate) [inline src/main/java/networkbook/logic/commands/OpenLinkCommand.java]: This can be changed to use assertion to get rid of runtime errors

**Review by Singa-pirate — CHANGES_REQUESTED** [review]
LGTM, except for a slight nitpick that can be improved

**Review by Singa-pirate — APPROVED** [review]
LGTM!

**Pull request merged**